Init User Management API project

trhuloc opened the pull request:
Implementation: User, Role model
Implementation: EntityFrameWork
Implementation: Code First Migrations
Implementation: Unit Of Work and Generic Repository

trhuloc committed:
Init User Management API project
Implementation: User, Role model
Implementation: EntityFrameWork
Adding a Migration

beannguyen commented on Simple-User-Management-API/Simple-User-Management-API/Models/UserManagementContext.cs:
```diff
@@ -0,0 +1,25 @@
+    {
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            optionsBuilder.UseSqlServer("Server=localhost;Database=UserManagementDB;Trusted_Connection=True;");
```
Move to configuration, use IOptions pattern
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-3.1

beannguyen commented on Simple-User-Management-API/Simple-User-Management-API/Startup.cs:
```diff
@@ -0,0 +1,51 @@
+    public class Startup
+    {
+        public Startup(IConfiguration configuration)
+        {
```
Enable CORS to allow FE access the API.

https://docs.microsoft.com/en-us/aspnet/core/security/cors?view=aspnetcore-3.1

voquanghien commented:
@trhuloc 
- Please move .gitignore to the folder which is the same level one that contains README.md, .git folder because I see you create a new folder for API so I assume that you could create more folders so one .gitignore in parent folder will handle all of folders/files
- Please rebase develop again, I have some update in develop
- One more thing. you should comment to explain codes which you think are necessary or hard to understand

trhuloc committed:
Resolve Comment:
Move config to configuration, use IOptions pattern
Enable CORS to allow FE access the API.
Move .gitignore to the folder which is the same level one that contains README.md

trhuloc committed:
Implementation Unit Of Work and Generic Repository
